File: go/testcode/benchmark_test_00360.go
package testcode

import (
	"fmt"
	"net/http"

	"github.com/go-ldap/ldap/v3"
)

func BenchmarkTest00360(w http.ResponseWriter, r *http.Request) {
	user := r.FormValue("user")
	pass := r.FormValue("pass")
	filter := fmt.Sprintf("(&(uid=%s)(userPassword=%s))", user, pass)
	searchReq := ldap.NewSearchRequest("dc=example,dc=com", ldap.ScopeWholeSubtree, ldap.NeverDerefAliases, 0, 0, false, filter, []string{"dn"}, nil)
	result, err := LDAPConn.Search(searchReq)
	if err != nil {
		http.Error(w, err.Error(), 500)
		return
	}
	authenticated := len(result.Entries) > 0
	RespondJSON(w, http.StatusOK, map[string]bool{"authenticated": authenticated})
}

File: ruby/testcode/benchmark_test_00360.rb
require_relative 'shared'

module Net
  module LDAP
    def self.new(options = {}); options; end
    def self.search(options = {}); []; end
    module Filter
      def self.eq(attr, value); "(#{attr}=#{value})"; end
    end
  end
end

def lookup_user_filtered(req)
  username = req.param('username')
  ldap = Net::LDAP.new(host: 'ldap.example.com')
  safe_filter = Net::LDAP::Filter.eq('uid', username)
  entries = ldap.search(filter: safe_filter)
  BenchmarkResponse.json({ count: entries.length })
end
